### PR TITLE
Dangerous default path would delete everything in project folder

### PR DESCRIPTION
**Pull request by m-abs**

### CODE REVIEW

File: js/tests.js [hunk 1]
```diff
@@ -83,6 +83,10 @@ Constructor.prototype.extract = function(relDest, stream, callback) {
     // TODO: Remove the tests anchored at dest
     callback(err);
   }
+  if ( !dest || dest === './' || dest === '.' ) {
```
Comment (mzedeler) [inline js/tests.js]: Please provide a better check here.

File: js/server.js [hunk 2]
```diff
@@ -10,12 +11,19 @@ var express = require('express'),
     Tests = require('./tests.js');
 
 var Constructor = function(path, uploadKey) {
-
   var mount = '/tests';
+
+  var cwd = pathResolve( process.cwd() );
+  path = pathResolve( path || cwd + mount );
+
+  if ( cwd.substr( 0, path.length ) === path ) {
```
Comment (mzedeler) [inline js/server.js]: I'd understand this easier if it was

```
cwd.indexOf(path) === 0
```

or

```
cwd.match('^' + path)
```

File: js/tests.js [hunk 3]
```diff
@@ -61,19 +63,21 @@ Constructor.prototype.parseManifest = function(manifestPath, callback) {
...
+      },
       callback
     );
   }
 };
 
 Constructor.prototype.extract = function(relDest, stream, callback) {
-  var join = require('path').join;
-  var dest = join(this.path, relDest);
+  var dest = pathResolve( join(this.path, relDest) );
```
Comment (mzedeler) [inline js/tests.js:80]: Why `pathResolve` here?

Comment (m-abs) [inline js/tests.js:80]: The absolute path is needed to check against CWD.

**Pull request merged**